### PR TITLE
Teach travis how to regenerate gh-pages.

### PR DESCRIPTION
**Pull request by shicks**
Adds a secure token and a script to generate gh-pages.  I've run the script manually (passing in what I believe Travis will pass for the various environment variables) and confirmed that it works.  Next step is to push to master and see if that side is correct.

### CODE REVIEW

File: scripts/ci/generate_latest_docs.sh [hunk 1]
```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+# Rebuild gh-pages branch, adapted from
+# https://github.com/google/dagger/blob/master/util/generate-latest-docs.sh
+
+HOME=/tmp/aaa
```
Comment (joeltine) [inline scripts/ci/generate_latest_docs.sh]: `$(mktmp)`?

File: scripts/ci/generate_latest_docs.sh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    --type_filter '^.*+_$'
+  )
+
+  while read -r file; do
+    file=${file#./}
+    if [[ "$file" =~ ^closure/goog/demos ]]; then continue; fi
+    if [[ "$file" =~ _test\.js$ || "$file" =~ _perf\.js$ ]]; then continue; fi
+    if [[ "${BLACKLIST[${file#closure/goog/}]}" == true ]]; then continue; fi
+    if [[ "$file" =~ ^closure/goog || "$file" =~ ^third_party/closure/goog ]]; then
+      command=("${command[@]}" --source "$file")
```
Comment (joeltine) [inline scripts/ci/generate_latest_docs.sh]: You really have to run the command on every single file? I thought dossier would just operate on a root directory.

File: scripts/ci/generate_latest_docs.sh [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  done < <(find . -name '*.js')
+
+  # Run dossier.
+  "${command[@]}"
+
+  # Make a commit and push it.
+  (
+    cd $HOME/gh-pages
+    git add -A
+    git commit -m "Latest documentation on successful travis build $TRAVIS_BUILD_NUMBER auto-pushed to gh-pages"
```
Comment (joeltine) [inline scripts/ci/generate_latest_docs.sh]: Include `$TRAVIS_COMMIT` too.

**Comment by concavelenz** [comment]
Nice

File: scripts/ci/generate_latest_docs.sh (hunk 1 above)
Comment (shicks) [inline scripts/ci/generate_latest_docs.sh]: Oops, that was just so I could test it on my local machine without dumping lots of crap directly into my home directory.  Might make sense just to do the whole thing in tmp regardless.

File: scripts/ci/generate_latest_docs.sh (hunk 2 above)
Comment (shicks) [inline scripts/ci/generate_latest_docs.sh]: It does, but we're filtering out various files in a way that is more complicated than what dossier can do by itself, presumably.

This is basically just adapted from the python script we use internally, complete with the filters and blacklist, etc.

File: scripts/ci/generate_latest_docs.sh (hunk 3 above)
Comment (shicks) [inline scripts/ci/generate_latest_docs.sh]: Done. Reworded slightly.

**Comment by shicks** [comment]
Thanks for the review, merging.

**Pull request merged**